aptos-move/framework/aptos-framework/sources/aptos_coin.move

```diff
@@ -150,7 +150,42 @@ module aptos_framework::aptos_coin {
     }
 
     #[test_only]
-    use aptos_framework::aggregator_factory;
+    public fun mint_fa_for_test(amount: u64): FungibleAsset acquires MintCapStore {
+        ensure_initialized_with_fa_metadata_for_test();
+        coin::coin_to_fungible_asset(
+            coin::mint(
+                amount,
+                &borrow_global<MintCapStore>(@aptos_framework).mint_cap
+            )
+        )
+    }
+
+    #[test_only]
+    public fun mint_fa_to_for_test<T: key>(store: Object<T>, amount: u64) acquires MintCapStore {
+        fungible_asset::deposit(store, mint_fa_for_test(amount));
+    }
+
+    #[test_only]
+    public fun mint_fa_to_primary_fungible_store_for_test(
+        owner: address,
+        amount: u64,
+    ) acquires MintCapStore {
+        primary_fungible_store::deposit(owner, mint_fa_for_test(amount));
+    }
+
+    #[test_only]
+    public fun ensure_initialized_with_fa_metadata_for_test() {
+        if (!exists<MintCapStore>(@aptos_framework)) {
+            let aptos_framework = account::create_signer_for_test(@aptos_framework);
+            if (!aggregator_factory::aggregator_factory_exists_for_testing()) {
+                aggregator_factory::initialize_aggregator_factory_for_test(&aptos_framework);
+            };
+            let (burn_cap, mint_cap) = initialize(&aptos_framework);
```

Comment:
```suggestion
        if (!exists<MintCapStore>(@aptos_fungible_asset)) {
            let aptos_framework = account::create_signer_for_test(@aptos_fungible_asset);
            if (!aggregator_factory::aggregator_factory_exists_for_testing()) {
                aggregator_factory::initialize_aggregator_factory_for_test(&aptos_fungible_asset);
            };
            let (burn_cap, mint_cap) = initialize(&aptos_fungible_asset);
```

Per https://github.com/aptos-labs/aptos-core/pull/11224#discussion_r1516706137, correct?

Reply:
correct. I made those changes and set `@aptos_fungible_asset = 0xA`. Let me know whether I missed anything. 
AFAIK, every test passes.

Reply:
Got it, yes this works. The `CoinType` and the `MintCapStore` will be stored under `@aptos_framework`, but the auto-generated FA metadata will be stored under `@aptos_fungible_asset` per `coin::ensure_paired_metadata()`. Thanks for clarifyig

Reply:
MintCapStore is still stored at 0x1 = @aptos_framework.